app/src/test/java/com/davidtomas/taskyapp/features/auth/data/remote/api/AuthServiceImplTest.kt

```diff
@@ -0,0 +1,133 @@
+package com.davidtomas.taskyapp.features.auth.data.remote.api
+
+import assertk.assertThat
+import assertk.assertions.isEqualTo
+import assertk.assertions.isTrue
+import com.davidtomas.taskyapp.core.domain._util.Result
+import com.davidtomas.taskyapp.core.domain._util.isError
+import com.davidtomas.taskyapp.core.domain._util.isSuccess
+import com.davidtomas.taskyapp.core.domain.model.DataError
+import com.davidtomas.taskyapp.features.auth.domain.useCase.LoginUseCase
+import com.davidtomas.taskyapp.features.auth.domain.useCase.RegisterUseCase
+import com.davidtomas.taskyapp.util.AddLogExtension
+import io.ktor.http.HttpStatusCode
+import io.ktor.utils.io.errors.IOException
+import io.mockk.spyk
+import kotlinx.coroutines.test.runTest
+import kotlinx.serialization.SerializationException
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.ExtendWith
+import org.junit.jupiter.params.ParameterizedTest
+import org.junit.jupiter.params.provider.MethodSource
+import org.koin.test.KoinTest
+
+@ExtendWith(AddLogExtension::class)
+internal class AuthServiceImplTest : KoinTest {
+
+    private lateinit var authService: AuthServiceImpl
+
+    @Test
+    fun `Login is executed with valid data, it returns success response`() =
+        runTest {
+            authService = spyk(AuthServiceImpl(client = ktorSuccessClient))
+            val result = authService.login(
+                LoginUseCase.LoginParams(
+                    email = "email",
+                    password = "password"
+                )
+            )
+            assertThat((result as Result.Success).data.token).isEqualTo("token")
+            assertThat(result.isSuccess()).isTrue()
+        }
+
+    @ParameterizedTest
+    @MethodSource("httpStatusCodes")
+    fun `When an error occurs in login, it returns DataError Network`(statusCode: HttpStatusCode) {
+        runTest {
+            authService =
+                spyk(AuthServiceImpl(client = ktorErrorClientWithSpecificError(statusCode)))
+            val result = authService.login(
+                LoginUseCase.LoginParams(
+                    email = "email",
+                    password = ""
+                )
+            )
+            val dataError = when (statusCode) {
+                HttpStatusCode.BadRequest.copy(409) -> DataError.Network.BAD_REQUEST
+                HttpStatusCode.InternalServerError -> DataError.Network.SERVER_ERROR
+                HttpStatusCode.Unauthorized -> DataError.Network.UNAUTHORIZED
+                else -> DataError.Network.UNKNOWN
```

Comment:
Could it make sense to make this HTTP to domain error mapping logic an extension function? Otherwise, there will be quite some repitition